### ISSUE TITLE
scrcpy --rotation=2  does not respect current orientation. It rotates the screen even when it is already rotated.

### ISSUE THREAD

**Issue by StacktraceException** [Genymobile/scrcpy#4219]
- [ ] I have read the [FAQ](https://github.com/Genymobile/scrcpy/blob/master/FAQ.md).
 - [ ] I have searched in existing [issues](https://github.com/Genymobile/scrcpy/issues).

**Environment**
 - OS: (sadly)  Windows (11).
 - scrcpy version: [e.g. 2.1.1 ]
 - installation method: [Windows release...]
 - device model: Xiaomi 11T
 - Android version: [e.g. 13]

**Describe the bug**

The problem is that by default the orientation is set to vertical, but to prevent this (as horizontal is better) I tried using --rotation=2. But this does rotates the screen even when it is already landscape so that it becomes upside down :/.

It needs to work properly regardless of the starting screen orientation so that rotation=2 is always landscape and not upside down.

On errors, please provide the output of the console (and `adb logcat` if relevant).

```
Please paste terminal output in a code block.
```

Please do not post screenshots of your terminal, just post the content as text instead.

**Comment by rom1v** [comment]
rotation=0 (the default) is always the natural orientation (portrait for a phone, landscape for a tablet). Each increments add a 90° rotation counterclockwIse.

**Comment by StacktraceException** [comment]
It seems that you have commented on what the rotation setting does while
not acknowledging the problem I described. Before replying any further,
please reread the entirety.

pon., 7 sie 2023, 00:40 użytkownik Romain Vimont ***@***.***>
napisał:

**Comment by rom1v** [comment]
> It needs to work properly regardless of the starting screen orientation so that rotation=2 is always landscape and not upside down.

Nope, `rotation=2` means "rotate the window by 180°", so if the original is portrait, it displays portrait upside-down, if it is landscape, it displays landscape upside-down.

What you want to set is probably the mirroring orientation (with `--lock-video-orientation`), not the window rotation: https://github.com/Genymobile/scrcpy/blob/master/doc/video.md#rotation

**Comment by rom1v** [comment]
> Actually, like I have repeated already, the problem is that landscape mode
needs to be enabled at start. The option of lock screen orientation will
make it not rotate when portrait mode is preferred.

Only if you pass `--lock-video-orientation` without argument or with argument `--lock-video-orientation=initial`.

If you pass `--lock-video-orientation=1`, it will force the orientation to be 90° ccw from the natural orientation.

**Comment by StacktraceException** [comment]
Are you writing that - - lock-video-orientation=1 will make it always
landscape at start but allow rotation as normal? Please let me understand.
If so, then why isn't this in the documentation? Regards

pon., 7 sie 2023, 20:21 użytkownik Romain Vimont ***@***.***>
napisał:
